tests/ballerina-tools-integration-test/src/test/java/org/ballerinalang/test/packaging/PackagingInitTestCase.java

```diff
@@ -0,0 +1,253 @@
+                ("hello/sayHello"));
+        Assert.assertEquals(response.getResponseCode(), 200, "Response code mismatched");
+        ballerinaServerForService.stopServer();
+    }
+
+    @Test(description = "Test creating a project with a service and main in different packages")
+    public void testInitWithMainServiceInDiffPackage() throws Exception {
+        Path projectPath = tempProjectDirectory.resolve("secondTestWithPackages");
+        Files.createDirectories(projectPath);
+
+        sourceFiles = new ArrayList<>();
+        packageDescFiles = new ArrayList<>();
+
+        sourceFiles.add(new SrcFile("foo", FileType.MAIN));
+        packageDescFiles.add(new PackageMdFile("foo", FileType.MAIN));
+
+        sourceFiles.add(new SrcFile("bar", FileType.SERVICE));
+        packageDescFiles.add(new PackageMdFile("bar", FileType.SERVICE));
+
+        InitHandler.initialize(projectPath, manifest, sourceFiles, packageDescFiles);
+
+        Assert.assertEquals(Files.exists(projectPath.resolve("foo").resolve("main.bal")), true);
+        Assert.assertEquals(Files.exists(projectPath.resolve("bar").resolve("hello_service.bal")), true);
+        Assert.assertEquals(Files.exists(projectPath.resolve("Ballerina.toml")), true);
+        Assert.assertEquals(Files.exists(projectPath.resolve("foo").resolve("tests").resolve("main_test.bal")), true);
+        Assert.assertEquals(Files.exists(projectPath.resolve("bar").resolve("tests").resolve("hello_service_test.bal")),
+                            true);
+        // Test for main
+        ballerinaServerForMain = new ServerInstance(serverZipPath);
+        String[] clientArgs = {"--sourceroot", projectPath.toString(), "foo"};
+        logLeecher = new LogLeecher("Hello World!");
+        ballerinaServerForMain.addLogLeecher(logLeecher);
+        ballerinaServerForMain.runMain(clientArgs, getEnvVariables(), "run");
+
+        // Test for service
+        ballerinaServerForService = ServerInstance.initBallerinaServer();
+        ballerinaServerForService.startBallerinaServer(projectPath.resolve("bar").resolve("hello_service.bal")
+                                                                  .toString());
+        HttpResponse response = HttpClientRequest.doGet(ballerinaServerForService.getServiceURLHttp("hello/sayHello"));
+        Assert.assertEquals(response.getResponseCode(), 200, "Response code mismatched");
+        ballerinaServerForService.stopServer();
+    }
+
+    @Test(description = "Test creating a project without going to interactive mode")
+    public void testInitWithoutGoingToInteractiveMode() throws Exception {
+        Path projectPath = tempProjectDirectory.resolve("testWithoutPackage");
+        Files.createDirectories(projectPath);
+
+        sourceFiles = new ArrayList<>();
+        packageDescFiles = new ArrayList<>();
+
+        sourceFiles.add(new SrcFile("", FileType.SERVICE));
+        packageDescFiles.add(new PackageMdFile("", FileType.SERVICE));
```

Comment:
Shall we create a constructor with not package name arg ? (the first arg)

Reply:
Fixed with 6ff8577